sched/pthread/pthread_join.c

```diff
@@ -138,6 +138,12 @@ int pthread_join(pthread_t thread, FAR pthread_addr_t *pexit_value)
     }
   else
     {
+      if (pjoin->detached)
+        {
+          leave_cancellation_point();
```

Comment:
`pthread_sem_give(&group->tg_joinsem);` is missing. The previous logic with `if` / `else` would perform clean-up properly